presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -848,6 +848,63 @@ public void testRow()
+        Session actualSession = Session.builder(getSession())
+                .setCatalog("hive")
+                .setCatalogSessionProperty("hive", "parquet_pushdown_filter_enabled", "true")
+                .build();
+
+        // Expected session is for the Java query runner.
+        // It does not support Parquet filter pushdown yet, so we have to explicitly disable it.
+        Session expectedSession = Session.builder(actualSession)
+                .setCatalogSessionProperty("hive", "parquet_pushdown_filter_enabled", "false")
+                .build();
+
+        // Generate temporary table name.
+        String tmpTableName = generateRandomTableName();
+
+        try {
+            // Create a Parquet table with decimal types and test data.
+            getExpectedQueryRunner().execute(expectedSession, String.format("CREATE TABLE %s (c0 DECIMAL(15,2), c1 DECIMAL(38,2)) WITH (format = 'PARQUET')", tmpTableName));
+            getExpectedQueryRunner().execute(expectedSession, String.format("INSERT INTO %s VALUES (DECIMAL '0', DECIMAL '0')", tmpTableName));
+            getExpectedQueryRunner().execute(expectedSession, String.format("INSERT INTO %s VALUES (DECIMAL '1.2', DECIMAL '3.4')", tmpTableName));
+            getExpectedQueryRunner().execute(expectedSession, String.format("INSERT INTO %s VALUES (DECIMAL '1000000.12', DECIMAL '28239823232323.57')", tmpTableName));
+            getExpectedQueryRunner().execute(expectedSession, String.format("INSERT INTO %s VALUES (DECIMAL '-542392.89', DECIMAL '-6723982392109.29')", tmpTableName));
+
+            assertQuery(actualSession, expectedSession, String.format("SELECT * FROM %s WHERE c0 > DECIMAL '1.1' and c1 < DECIMAL '5.2'", tmpTableName));
+            assertQuery(actualSession, expectedSession, String.format("SELECT * FROM %s WHERE c0 >= DECIMAL '1.2' and c1 <= DECIMAL '5.2'", tmpTableName));
+            assertQuery(actualSession, expectedSession, String.format("SELECT * FROM %s WHERE c0 = DECIMAL '1.2' and c1 = DECIMAL '3.4'", tmpTableName));
+            assertQuery(actualSession, expectedSession, String.format("SELECT * FROM %s WHERE c0 > DECIMAL '-542392.89' and c1 <= DECIMAL '28239823232323.57'", tmpTableName));
+            assertQuery(actualSession, expectedSession, String.format("SELECT * FROM %s WHERE c0 >= DECIMAL '1.2'", tmpTableName));
+            assertQuery(actualSession, expectedSession, String.format("SELECT * FROM %s WHERE c1 < DECIMAL '5.2'", tmpTableName));
+            assertQuery(actualSession, expectedSession, String.format("SELECT * FROM %s WHERE c1 <= DECIMAL '3.4'", tmpTableName));
+            assertQuery(actualSession, expectedSession, String.format("SELECT * FROM %s WHERE c0 = DECIMAL '1.2'", tmpTableName));
+            assertQuery(actualSession, expectedSession, String.format("SELECT * FROM %s WHERE c1 = DECIMAL '3.4'", tmpTableName));
+
+            // Test min/max decimal values.
+            String shortDecimalMin = "DECIMAL '-999999999999999999'";
```

Comment:
Do these convert into range filters also ?

Reply:
It depends. Presto seems to always convert the lower and upper bounds as the largest value for the type. For example, DECIMAL(3,1) value would have an upper bound of 99.9 instead of 999999999999999999. If the types match, then it will be 999999999999999999.